### PR TITLE
feat: add MenuSeparator

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#73773

### CODE REVIEW

**Comment by gca-axelor** [comment]
test in any index by add menuentry with propr separator = true


    stock_menu_separator: {
      title: 'Stock_Separator',
      separator: true,
    },

File: packages/core/src/navigator/drawer/MenuItem.js [hunk 1]
```diff
@@ -56,6 +61,7 @@ const MenuItem = ({
       disabled={menuItem.disabled || disabled}
       compatibility={menuItem.compatibilityAOS}
       isActive={isActive}
+      separator={menuItem.separator}
```
Comment (vhu-axelor) [inline packages/core/src/navigator/drawer/MenuItem.js:64]: Why did you add this props here? MenuItemEntry doesn't seem to use it

File: packages/core/src/navigator/drawer/MenuSeparator.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const Colors = useThemeColor();
+
+  const styles = useMemo(() => {
+    return getStyles(Colors);
+  }, [Colors]);
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.childrenContainer}>
+        <View style={[styles.menuItemActive]} />
```
Comment (vhu-axelor) [inline packages/core/src/navigator/drawer/MenuSeparator.js:33]: I think you can remove it and thus delete the View with childrenContainer style, don't forget to remove styles of deleted components

File: packages/core/src/navigator/drawer/MenuSeparator.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        <View style={[styles.menuItemContainer, style]}>
```
Comment (vhu-axelor) [inline packages/core/src/navigator/drawer/MenuSeparator.js:34]: If you want to put style props here maybe rename it to make it clear that it allows you to modify this part and not the entire component

File: packages/core/src/navigator/drawer/MenuSeparator.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      width: '100%',
+    },
+    menuItemActive: {
+      width: 7,
+      height: 32,
+      borderTopRightRadius: 8,
+      borderBottomRightRadius: 8,
+    },
+    menuItemIcon: {
+      marginLeft: 0,
```
Comment (vhu-axelor) [inline packages/core/src/navigator/drawer/MenuSeparator.js:71]: I think it's not useful

File: packages/core/src/navigator/drawer/MenuSeparator.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      <View style={styles.childrenContainer}>
+        <View style={[styles.menuItemActive]} />
+        <View style={[styles.menuItemContainer, style]}>
+          <Icon
+            style={styles.menuItemIcon}
+            name={'chevron-right'}
+            size={iconSize}
+            color={Colors.secondaryColor.background}
+          />
+          <View style={styles.menuItemTextContainer}>
```
Comment (vhu-axelor) [inline packages/core/src/navigator/drawer/MenuSeparator.js:41]: Remove it and put style in menuItemContainer

**Comment by vhu-axelor** [comment]
I forgot to put it in my review but the color gives the impression that the entry is disabled

**Review by vhu-axelor — APPROVED** [review]

File: packages/core/src/app/Module.ts [hunk 6]
```diff
@@ -37,7 +38,7 @@ interface MenuWithSubMenus extends MenuBase {
 }
 
 interface MenuWithScreen extends MenuBase {
-  screen: string;
+  screen?: string;
```
Comment (lme-axelor) [inline packages/core/src/app/Module.ts:41]: No, that's not logical to have a type name MenuWithScreen and to mark screen as optionnal. You need to create a new type MenuSeparator which should contains only the props taken into account in your separator and then Menu type should be  MenuWithSubMenus | RootMenuWithScreen | MenuSeparator.
Try to factorize the type to avoid duplicated code.
I think separator should take title, hideIf, parent, order and separator props into account

File: packages/core/src/navigator/drawer/MenuSeparator.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  const styles = useMemo(() => {
+    return getStyles(Colors);
+  }, [Colors]);
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.menuItemContainer}>
+        <Icon
+          style={styles.menuItemIcon}
+          name={'chevron-right'}
```
Comment (lme-axelor) [inline packages/core/src/navigator/drawer/MenuSeparator.js:35]:
```
          name='chevron-right'
``` 
is cleaner

File: packages/core/src/navigator/drawer/MenuSeparator.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+          size={iconSize}
+          color={Colors.secondaryColor.background}
+        />
+        <Text style={styles.menuItemTitle}>{title}</Text>
```
Comment (lme-axelor) [inline packages/core/src/navigator/drawer/MenuSeparator.js:39]: Do not create a style just for fontSize and text color, you have props for that.
Also, leave default text color for the title
